ambry-server/src/integration-test/java/com.github.ambry.server/ServerTestUtil.java

```diff
@@ -879,41 +879,37 @@ protected static void endToEndReplicationWithMultiNodeSinglePartitionTest(String
+      ArrayList<byte[]> encryptionKeyList = new ArrayList<>();
       byte[] usermetadata = new byte[1000];
-      byte[] data = new byte[1000];
-      short accontId = Utils.getRandomShort(TestUtils.RANDOM);
-      short containerId = Utils.getRandomShort(TestUtils.RANDOM);
-      BlobProperties properties = new BlobProperties(1000, "serviceid1", accontId, containerId);
-      new Random().nextBytes(usermetadata);
-      new Random().nextBytes(data);
+      TestUtils.RANDOM.nextBytes(usermetadata);
       PartitionId partition = clusterMap.getWritablePartitionIds().get(0);
-      BlobId blobId1 = new BlobId(BlobId.DEFAULT_FLAG, clusterMap.getLocalDatacenterId(), properties.getAccountId(),
-          properties.getContainerId(), partition);
-      BlobId blobId2 = new BlobId(BlobId.DEFAULT_FLAG, clusterMap.getLocalDatacenterId(), properties.getAccountId(),
-          properties.getContainerId(), partition);
-      BlobId blobId3 = new BlobId(BlobId.DEFAULT_FLAG, clusterMap.getLocalDatacenterId(), properties.getAccountId(),
-          properties.getContainerId(), partition);
-      BlobId blobId4 = new BlobId(BlobId.DEFAULT_FLAG, clusterMap.getLocalDatacenterId(), properties.getAccountId(),
-          properties.getContainerId(), partition);
-      BlobId blobId5 = new BlobId(BlobId.DEFAULT_FLAG, clusterMap.getLocalDatacenterId(), properties.getAccountId(),
-          properties.getContainerId(), partition);
-      BlobId blobId6 = new BlobId(BlobId.DEFAULT_FLAG, clusterMap.getLocalDatacenterId(), properties.getAccountId(),
-          properties.getContainerId(), partition);
-      BlobId blobId7 = new BlobId(BlobId.DEFAULT_FLAG, clusterMap.getLocalDatacenterId(), properties.getAccountId(),
-          properties.getContainerId(), partition);
-      BlobId blobId8 = new BlobId(BlobId.DEFAULT_FLAG, clusterMap.getLocalDatacenterId(), properties.getAccountId(),
-          properties.getContainerId(), partition);
-      BlobId blobId9 = new BlobId(BlobId.DEFAULT_FLAG, clusterMap.getLocalDatacenterId(), properties.getAccountId(),
-          properties.getContainerId(), partition);
-      BlobId blobId10 = new BlobId(BlobId.DEFAULT_FLAG, clusterMap.getLocalDatacenterId(), properties.getAccountId(),
-          properties.getContainerId(), partition);
-      BlobId blobId11 = new BlobId(BlobId.DEFAULT_FLAG, clusterMap.getLocalDatacenterId(), properties.getAccountId(),
-          properties.getContainerId(), partition);
+
+      for (int i = 0; i < 11; i++) {
+        short accountId = Utils.getRandomShort(TestUtils.RANDOM);
+        short containerId = Utils.getRandomShort(TestUtils.RANDOM);
+        propertyList.add(new BlobProperties(1000, "serviceid1", accountId, containerId, true));
+        blobIdList.add(
+            new BlobId(BlobId.DEFAULT_FLAG, clusterMap.getLocalDatacenterId(), accountId, containerId, partition));
+        dataList.add(TestUtils.getRandomBytes(1000));
+        boolean isEncrypted = TestUtils.RANDOM.nextBoolean();
```

Comment:
aren't you setting `encrypted` to `true` in all created `BlobProperties` in line 893?